earthaccess/search.py

```diff
@@ -11,6 +11,33 @@
 from .results import DataCollection, DataGranule
 
 
+def _normalize_datetime(raw: None | str | dt.date | dt.datetime) -> None | dt.datetime:
+    # empty string or None is None
+    if not raw:
+        return None
+    # the cmr.*Query.temporal method only parses a subset of ISO 8601 strings, so we
+    # add flexibility, inclusive of the presence or absence of timezone information
+    if isinstance(raw, str):
+        normalized = dateutil.parser.parse(raw)
+    elif not isinstance(raw, dt.datetime):
+        normalized = dt.datetime.combine(raw, dt.time())
+    else:
+        normalized = raw
+    # and then we guarantee a UTC datetime, assuming that a naive datetime
+    # object IS utc (thus IS NOT local time, because science)
+    try:
+        tz = normalized.tzinfo
+    except AttributeError:
+        raise TypeError(
+            f"Dates must be date, datetime or str, not {dt.__class__.__name__}."
```

Comment:
```suggestion
            f"Dates must be date, datetime or str, not {raw.__class__.__name__}."
```
This should be `raw`, no?